runtime/src/main/java/org/corfudb/runtime/object/MVOCache.java

```diff
@@ -0,0 +1,279 @@
+package org.corfudb.runtime.object;
+
+import com.google.common.base.Preconditions;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalCause;
+import com.google.common.cache.RemovalNotification;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+import org.corfudb.runtime.CorfuRuntime;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+/**
+ * MVOCache is the centralized container that holds the reference to
+ * all MVOs. Overall it provides put and get APIs and manages the
+ * Cache-related properties (LRU) under the hood.
+ *
+ */
+@Slf4j
+public class MVOCache<T extends ICorfuSMR<T>> {
+
+    private final CorfuRuntime runtime;
+
+    // A registry to keep track of all opened MVOs
+    @Getter
+    private final ConcurrentHashMap<UUID, MultiVersionObject<T>> allMVOs = new ConcurrentHashMap<>();
```

Comment:
We should also have an API to remove objects after they are opened. Some applications want to open the table once and then release it from memory.